### ISSUE TITLE
Potentially always incorrect warning?

### ISSUE THREAD

**Issue by cameron-elliott** [golang/lint#466]
This function:

```go
func X(d []byte) bool {
	return d != nil && len(d)>0
}
```

Always generates this warning:

> should omit nil check; len() for nil slices is defined as zero (S1009)

My thought is this, while the warning is true for `d==nil && len(d) `
I think the warning is erroneous in the case where you want to check for non-nil and non-zero slice status.

Thoughts?

**Comment by ainar-g** [comment]
<p>
Not sure what you mean here. Any slice that has a <code>len</code>
greater than zero is a non-nil slice. And the condition in the code you
provided is redundant, as <code>len(nil) == 0</code>.
</p>

**Comment by cameron-elliott** [comment]
You are absolutely right. len(x)>0 is sufficient for a non-nil check also.
